### PR TITLE
Reimplement compact-shard command using tsm1.Compactor

### PR DESCRIPTION
**Pull request by stuartcarnie**
* rename `influx-tools` to `influx_tools`
* reimplement `compact-shard` command using `tsm1.Compactor` to fully compact the specified shard directory

### CODE REVIEW

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/166531770/src/github.com/influxdata/influxdb/cmd/influx_tools/compact/command.go:41:2: field store is unused (U1000)
/tmp/166531770/src/github.com/influxdata/influxdb/cmd/influx_tools/compact/command.go:47:2: field count is unused (U1000)
/tmp/166531770/src/github.com/influxdata/influxdb/cmd/influx_tools/compact/command.go:48:3: field series is unused (U1000)
/tmp/166531770/src/github.com/influxdata/influxdb/cmd/influx_tools/compact/command.go:49:3: field values is unused (U1000)
```

**Comment by e-dard** [comment]
@stuartcarnie not sure what's going on with the bot. The error is:

```
cmd/influx_tools/compact/command.go:41:2: field store is unused (U1000)
```

File: cmd/influx_tools/compact/command.go [hunk 1]
```diff
@@ -62,39 +61,26 @@ func (cmd *Command) Run(args []string) (err error) {
...
+		if err != nil {
+			return err
+		}
 	}
 
-	dataPath := filepath.Join(cmd.server.TSDBConfig().Dir, cmd.database, cmd.rp)
-	shardDataPath := filepath.Join(dataPath, strconv.Itoa(int(cmd.shardID)))
-	walPath := filepath.Join(cmd.server.TSDBConfig().WALDir, cmd.database, cmd.rp)
-	shardWalPath := filepath.Join(walPath, strconv.Itoa(int(cmd.shardID)))
+	fmt.Printf("opening shard at path %q\n\n", cmd.path)
```
Comment (e-dard) [inline cmd/influx_tools/compact/command.go]: I think maybe we should use `fmt.FprintX` with `cmd.Stdout` so it's easier to add options to change the output.

File: cmd/influx_tools/compact/command.go [hunk 2]
```diff
@@ -109,231 +95,222 @@ func (cmd *Command) Run(args []string) (err error) {
...
-		fmt.Printf("Compaction failed: unable to replace files\n%v", err)
-		return errors.New("unable to replace files")
+		return nil, fmt.Errorf("error reading tombstone files: %v", err)
 	}
 
-	fmt.Println("Compaction succeeded. New files:")
-	for _, f := range newFiles {
-		fmt.Printf("  %s\n", f)
+	fs.readers = make([]*tsm1.TSMReader, 0, len(fs.tsm))
+	err = fs.openFiles()
```
Comment (e-dard) [inline cmd/influx_tools/compact/command.go]: nit, can you collapse this into `if err = fs.openFiles(); err != `

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
``````

**Pull request merged**